src/jeluard/boot_notify.clj

```diff
@@ -51,22 +51,18 @@
   (let [title (or title "Boot notify")
         base-message {:title title :pid (or pid title)
                       :icon (or icon (boot-logo))}
-        messages (merge {:success "Success!" :warning "Got %s warnings during compilation" :failure "Failed to compile"} template)]
+        messages (merge {:success "Success!" :warning "%s warning/s" :failure "%s"} template)]
     (if-let [n (or notifier default-notifier)]
       (if (-supported? n)
-        (core/with-post-wrap
-          fileset
-          (try
-            (let [w @core/*warnings*]
-                 (if (zero? w)
-                   (when-let [s (:success messages)]
-                     (notify! n s base-message))
-                   (when-let [s (:warning messages)]
-                     (notify! n (format s w) base-message)))
-                 fileset)
-            (catch Throwable t
-              (when-let [s (:failure messages)]
-                (notify! n s base-message))
-              (throw t))))
+        (fn [next-task]
+          (fn [fileset]
+            (try
+              (util/with-let [_ (next-task fileset)]
+                (if (zero? @core/*warnings*)
+                  (notify! n (:success messages) base-message)
```

Comment:
@jeluard I thought the [type signature of that option](https://github.com/jeluard/boot-notify/blob/d779c5f961b7f5037342d0fb16cb279be4e0a8d4/src/jeluard/boot_notify.clj#L47) prevents non-string values.

```clojure
(notify :template {:success nil})
;=> java.lang.IllegalArgumentException: option :template must be of type {kw str}
```